pkg/exporter/sumologicexporter/exporter.go

```diff
@@ -397,7 +397,7 @@ func (se *sumologicexporter) pushMetricsData(ctx context.Context, md pdata.Metri
 				}
 
 				// If metadata differs from currently buffered, flush the buffer
-				if currentMetadata.string() != previousMetadata.string() && previousMetadata.string() != "" {
+				if !currentMetadata.equals(previousMetadata) && !previousMetadata.isEmpty() {
```

Comment:
This wouldn't work. You cannot use `continue` like this. I understand the condition would be easier to read if it was positive instead of negative, but I don't see a good way to do this.

Reply:
do not trust github :confused: 
![image](https://user-images.githubusercontent.com/58699848/156390785-8a542516-02d7-4ec2-9783-9bcf85a66d89.png)